### ISSUE TITLE
Accuracy metrics returning 0 with dense features

### ISSUE THREAD

**Issue by giuliodavide** [massquantity/LibRecommender#84]
Hello,
I have been using your librecommender for my MSc thesis involving audio features. The audio features are continuous dense features such as mfcc, i-vector commonly used in music research. I noticed once using librecommender, for features like genre, artists I can recieve meaningful results for accuracy metrics, but when I use audio features the system returns me zero accuracy for any of the audio features I use. So I was thinking that I am using something incorrectly, but I cannot figure out what. 
Here is how I feed my features in the system:

''' data is an object I created, in data.data we will find the pandas DataFrame containing the dataset '''
sparse_col = []
dense_col = list(data.data.columns)[4:]  # vector features of mfcc in this case
user_col = []
item_col = list(data.data.columns)[4:]  # vector features of mfcc in this case

''' data are already splitted in train and test, so I don't use your splitting functions '''
train_data, eval_data = data.data.merge(mfcc_train.loc[:, 'item'], on='item'), data.data.merge(mfcc_test.loc[:, 'item'], on='item') 

DatasetFeat.reset_feature_state()

train_data, data_info = DatasetFeat.build_trainset(
    train_data=train_data,
     user_col=user_col, 
     item_col=item_col, 
     sparse_col=sparse_col, 
     dense_col=dense_col,
    )

eval_data = DatasetFeat.build_testset(eval_data)

''' data are implicit so I build negative samples '''
train_data.build_negative_samples(data_info, item_gen_mode='random',
                                  num_neg=1, seed=2020)
eval_data.build_negative_samples(data_info, item_gen_mode='random',
                                  num_neg=1, seed=2222)

metrics = ['precision', 'recall', 'map', 'ndcg']
task = 'ranking'
print(data_info)

tf.compat.v1.reset_default_graph()
print(f'FM {feature}')
fm = FM(task, data_info, embed_size=16, n_epochs=3,
        lr=1e-4, lr_decay=False, reg=None, batch_size=256,
        num_neg=1, use_bn=True, dropout_rate=None, tf_sess_config=None)
fm.fit(train_data, verbose=2, shuffle=True)  ''' I don't set metrics parameter because I want to save the results later '''

eval_row = evaluate(model=fm, data=eval_data, metrics=metrics)  ''' here I save evaluation results '''

**Comment by massquantity** [comment]
It seems that there is nothing wrong with your approach. My gut feeling is that pure dense features may not be suitable for FM model, so you can try other models in this library or tree models such as [GradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html)
